### ISSUE TITLE
Null response when type returned by interface doesn't match fragment

### ISSUE THREAD

**Issue by Mithras** [apollographql/router#1286]
**Describe the bug**
When type returned by an interface field doesn't match any non-inline fragments, router returns `null`. This means that specifying multiple non-inline fragments on interface fields will **always** result in `null` response from the router.  Inline fragments work as expected, only non-inline fragments are affected.

**To Reproduce**
```graphql
fragment FragmentA on MyTypeA {
  something
}

fragment FragmentB on MyTypeB {
  somethingElse
}

query ThisDoesNotWork {
  # returns MyTypeA
  myInterfaceField {
    id
    ...FragmentB # BUG: the whole response is null because of this
  }
}

query ThisWorks {
  # returns MyTypeA
  myInterfaceField {
    id
    ...FragmentA # when fragment matches retuned type, everything works fine
  }
}

query ThisAlsoWorks {
  # returns MyTypeA
  myInterfaceField {
    id
    # inline fragments work are not affected and this works as expected
    ... on MyTypeA {
      something
    }
    ... on MyTypeB {
      somethingElse
    }
  }
}
```
The first query (with `...FragmentB`) always returns `null`:
```json
{
  "data": null
}
```

**Expected behavior**
Multiple non-inline fragments work as expected.

**Desktop (please complete the following information):**
 - OS: ubuntu 20.04
 - Version: ghcr.io/apollographql/router:v0.9.5

**Additional context**
At some point fragments on interfaces didn't work at all but that was fixed in https://github.com/apollographql/router/issues/928
It seems like something is still not quite right with interfaces and non-inline fragments.

**Comment by Geal** [comment]
thanks for the report, I'll dig into it

**Comment by Mithras** [comment]
Yes, I think it is a non-null list of non-null entities. There is no shape, just { data: null }. Let me know if you need more details

**Comment by Geal** [comment]
ok, if it's a non null, it makes sense (otherwise, the data would not be null but end up with null `somethingElse` field). It is fixed by #1295
